auto-wcag/auto-wcag
# Handle `content-visibility` and `hidden="until-found"`

Parking as draft for now.
Both `content-visibility` and `hidden="until-found"` have poor browser support. This means at least that any test case using them would hit an Accessibility Support problem…

---

* Add `content-visibility: hidden` to the reasons for being Programmatically Hidden.
* Add a not that `hidden: until-found` is recommended to be handled via `content-visibility: hidden`.
* Add inapplicable examples to rules that directly use "Programmatically Hidden".

There are many more rules that indirectly use "Programmatically hidden", because it is mentioned in "Included in the Accessibility tree". And indeed, programmatically hidden elements meet the conditions for being [excluded from the Accessibility Tree](https://www.w3.org/TR/wai-aria-1.2/#tree_exclusion).

I'm however not sure we need to add examples with these new properties or value on half the ruleset 🤔 
OTOH, it may be worth doing it like here with one single Inapplicable Example grouping many cases of exclusion. This does not clutter rules with a lot of extra examples, but still allow to exercice the definition (and it becomes the task of the tool vendors to understand why they fail a given test cases)

Closes issue(s):

- closes #1860 

Need for Call for Review:
This will require a 2 weeks Call for Review (updating a definition with impact in many rules)

---

## Pull Request Etiquette

### **When creating PR:**


### **After creating PR:**


### **When merging a PR:**


## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.
f
